### ISSUE TITLE
HLint is unaware of unlifted types

### ISSUE THREAD

**Issue by phadej** [ndmitchell/hlint#1199]
```
Found:
  data SmallMutableArray s a
    = SmallMutableArray (SmallMutableArray# s a)
Perhaps:
  newtype SmallMutableArray s a
    = SmallMutableArray (SmallMutableArray# s a)
```

**Comment by googleson78** [comment]
Hm, there's even a test for this!

**Comment by phadej** [comment]
I use hlint-3.2, is it fixed recently?

**Comment by googleson78** [comment]
Compiling `hlint` right now to check it out, but it shouldn't be a "recent" thing. Maybe some bug related to your example having more type variables..

**Comment by googleson78** [comment]
Ah, yes, I think it was recently fixed, it doesn't trigger from master.
Here's what seems to be the relevant issue - https://github.com/ndmitchell/hlint/issues/1166

**Comment by phadej** [comment]
Yes, no complains after upgrade. Sorry for the noise.
